pkg/server/server.go

```diff
@@ -0,0 +1,118 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	database "github.com/sachintiptur/http-app/pkg/util"
+)
+
+// DatabaseInterfaceStruct holds the database interface type
+type DatabaseInterfaceStruct struct {
```

Comment:
This is a bad struct name. Considering the code in this file, doesn't it look like this structure follows a common design pattern?